Sources/PubNub/PubNub.swift

```diff
@@ -100,7 +100,7 @@ public extension PubNub {
     /// Apple Push Notification Service
     case apns
     /// Firebase Cloude Messaging
-    case gcm
+    case fcm = "gcm"
```

Comment:
I wanted to replace occurrences of `gcm` with `fcm`. The reason why it's like that is because the value from the enum case is used as a value for the `type` query parameter in [this](https://www.pubnub.com/docs/sdks/rest-api/list-add-remove-registrations-for-device) request. The server side requires us to pass either `apns`, `mpns` or... `gcm` and that's why I had to solve it like this.

This is also a breaking change for customer, they have to replace this code (`of:` parameter only):
```
pubnub.addPushChannelRegistrations([channel], for: token, of: .gcm) { ... }
```
With:
```
pubnub.addPushChannelRegistrations([channel], for: token, of: .fcm) { ... }

```
On the one hand, someone might say we shouldn't introduce breaking changes like this. But on the other hand, I think we should no longer stick to `gcm` naming. What's your take on that guys?

Reply:
in JVM in the enum we have both FCM and GCM and they both map to "gcm" in the request, as required by server. would that be possible (at least for a few versions until the next major release when you remove GCM)?

Reply:
Now I see you point in kotlin there is:
```
enum class PNPushType(s: String) {
    APNS("apns"),
    MPNS("mpns"),
    GCM("gcm"),
    FCM("gcm"),
    APNS2("apns2"),
    ;
```

I am not agains braking changes. This time we decided not to introduce them by providing new payload structure (FCMPayloadV2) and add deprecation warning on old structure that we will use in the future eventually.

Reply:
I could modify `PushService` by getting rid of `: String` in its declaration and provide `value()` method like that. However, this is also a breaking change, because customers won't be able to call built-in `rawValue` getter in their code:

```
/// The identifier of the Push Service being used
enum PushService: Codable, Hashable {
  @available(*, unavailable, renamed: "fcm")
  case gcm
  /// Apple Push Notification Service
  case apns
  /// Firebase Cloude Messaging
  case fcm
  /// Microsoft Push Notification Service
  case mpns
    
  func value() -> String {
    switch self {
    case .gcm, .fcm:
      return "gcm"
    case .apns:
      return "apns"
    case .mpns:
      return "mpns"
    }
  }
}
```

Unfortunately, I can't declare `PushService` as below. The compiler raises an error that the raw value for enum is not unique even though `gcm` was marked as unavailable:

```
enum PushService: String, Codable, Hashable {
  @available(*, unavailable, renamed: "fcm")
  case gcm
  /// Apple Push Notification Service
  case apns
  /// Firebase Cloude Messaging
  case fcm = "gcm"
  /// Microsoft Push Notification Service
  case mpns
}
```

Reply:
then I guess you could either leave it as "gcm" for now since it doesn't affect functionality or make a breaking-change to "fcm", up to you